File: 2020/day02b.py
def main():
    lines = open('input.txt', 'r').read().splitlines()
    correct = 0
    for line in lines:
        a = int(line[:line.index('-')]) - 1
        b = int(line[line.index('-') + 1 : line.index(' ')]) - 1
        target = line[line.index(' ') + 1]
        password = line[line.index(':') + 2 :]
        if password[a] != password[b] == target or password[b] != password[a] == target:
            correct += 1
    print(correct)

if __name__ == '__main__':
    main()

File: 2022/day02b.jl
function main()
    lines = readlines(open("input.txt"))
    score = 0

    for line in lines
        win_type = codepoint(line[3]) - 88
        opponent_move = codepoint(line[1]) - 65

        score += 3 * win_type
        score += mod(opponent_move + win_type - 1, 3) + 1
    end
    
    print("$score\n")
end

main()
